File: GiocoPython/menu.py
import pygame
from globalFunction import *

imgCrazyGoose = pygame.image.load("./images/scrittaCrazyGoose_700x230_noBagliore.png")


class Menu():
    def __init__(self, game):
        self.game = game
        #Mi salvo le coordinate del centro dello schermo
        self.mid_w, self.mid_h = self.game.DISPLAY_W / 2, self.game.DISPLAY_H / 2
        self.run_display = True
        self.cursor_rect = pygame.Rect(0, 0, 0, 0)
        self.offset = - 100


    #disegna il cursore quindi se bisogna modificarlo si fa qua
    def draw_cursor(self):
        draw_text(self.game, "*", 30, self.game.WHITE, self.cursor_rect.x, self.cursor_rect.y)


    #sovrappone, cioè disegna tutto ciò che c'è da disegnare
    # (a partire dall'angolo in alto a sx, coord. x=0, y=0)
    def blit_screen(self):
        self.game.window.blit(self.game.display, (0, 0))
        pygame.display.update()
        self.game.reset_keys()


class MainMenu(Menu):
    def __init__(self, game):
        Menu.__init__(self, game)
        self.state = "Start"
        self.startx, self.starty = self.mid_w, self.mid_h + 0
        #self.optionsx, self.optionsy = self.mid_w, self.mid_h + 50
        self.creditsx, self.creditsy = self.mid_w, self.mid_h + 90
        self.quitx, self.quity = self.mid_w, self.mid_h + 170
        self.cursor_rect.midtop = (self.startx + self.offset, self.starty)

    
    def display_menu(self):
        #Mostra il menu e richiama il metodo che controlla se ci sono eventi
        self.run_display = True
        while self.run_display:
            #Per non occupare moolta CPU abbasso gli fps (in pratica fa meno giri al sec xke aspetta un attimo ogni volta)
            pygame.time.wait(100)
            
            self.game.check_events()
            self.check_input()
            #Rimpe tutto lo schemo di Azzurro (sovvrascrivendo tutto ciò che c'era di disegnato prima)
            self.game.display.fill(self.game.AZZURRO)

            self.game.display.blit(imgCrazyGoose, (160, 50))

            draw_text(self.game, "Start Game", 30, self.game.WHITE, self.startx, self.starty)
            draw_text(self.game, "Credits", 30, self.game.WHITE, self.creditsx, self.creditsy)
            draw_text(self.game, "Quit", 30, self.game.WHITE, self.quitx, self.quity)
            
            
            self.draw_cursor()
            self.blit_screen()


    #Metodo per muovere su e giù il cursore
    def move_cursor(self):
        if self.game.DOWN_KEY:
            if self.state == 'Start':
                self.cursor_rect.midtop = (self.creditsx + self.offset, self.creditsy)
                self.state = 'Credits'
            elif self.state == 'Credits':
                self.cursor_rect.midtop = (self.quitx + self.offset, self.quity)
                self.state = 'Quit'
            elif self.state == 'Quit':
                self.cursor_rect.midtop = (self.startx + self.offset, self.starty)
                self.state = 'Start'
        elif self.game.UP_KEY:
            if self.state == 'Start':
                self.cursor_rect.midtop = (self.quitx + self.offset, self.quity)
                self.state = 'Quit'
            elif self.state == 'Credits':
                self.cursor_rect.midtop = (self.startx + self.offset, self.starty)
                self.state = 'Start'
            elif self.state == 'Quit':
                self.cursor_rect.midtop = (self.creditsx + self.offset, self.creditsy)
                self.state = 'Credits'


    #Controlla quale menu è stato scelto
    def check_input(self):
        self.move_cursor()
        if self.game.START_KEY:
            if self.state == 'Start':
                self.game.playing = True
                self.game.curr_menu = None
            elif self.state == 'Credits':
                self.game.curr_menu = self.game.credits
            elif self.state == 'Quit':
                #CHIUDE LA FINESTRA E TERMINA L'ESECUZIONE DEL PROGRAMMA
                pygame.quit()
                exit()
            
            #Se ha premuto invio di certo andrà in un altro menu/entrerà
            # nel gioco/chiuderà la finestra, quindi questo menu non deve più essere mostrato
            self.run_display = False


class CreditsMenu(Menu):
    def __init__(self, game):
        Menu.__init__(self, game)


    #schermata dei Credits
    def display_menu(self):
        self.run_display = True
        while self.run_display:
        	#Per non occupare moolta CPU abbasso gli fps (in pratica fa meno giri al sec xke aspetta un attimo ogni volta)
            pygame.time.wait(100)
        
            self.game.check_events()
            if self.game.START_KEY or self.game.BACK_KEY:
                self.game.curr_menu = self.game.main_menu
                self.run_display = False
            self.game.display.fill(self.game.AZZURRO)
            draw_text(self.game, 'Credits', 50, self.game.WHITE, self.mid_w, self.mid_h - 180)
            draw_text(self.game, 'Made by Crazy Goose Team:', 30, self.game.WHITE, self.mid_w, self.mid_h - 120)
            draw_text(self.game, 'Bellone Giulio', 30, self.game.WHITE, self.mid_w, self.mid_h - 30)
            draw_text(self.game, 'Calzia Mattia', 30, self.game.WHITE, self.mid_w, self.mid_h + 20)
            draw_text(self.game, 'Dastrù Alessandro', 30, self.game.WHITE, self.mid_w, self.mid_h + 70)
            draw_text(self.game, 'Durante Andrea', 30, self.game.WHITE, self.mid_w, self.mid_h + 120)
            draw_text(self.game, 'Hila Kledi', 30, self.game.WHITE, self.mid_w, self.mid_h + 170)
            draw_text(self.game, 'Hu Qiyan', 30, self.game.WHITE, self.mid_w, self.mid_h + 220)
            self.blit_screen()

File: GiocoPython/GiocoPython_v1.1/webApp/menu.js
class Menu {
    constructor(game, ctx) {
        this.ctx = ctx
        this.game = game
            //Mi salvo le coordinate del centro dello schermo
        this.mid_w = this.game.DISPLAY_W / 2
        this.mid_h = this.game.DISPLAY_H / 2
        this.run_display = true
        this.cursorX = 0
        this.cursorY = 0
            //this.cursor_rect = pygame.Rect(0, 0, 0, 0)
        this.offset = -100
    }

    draw_cursor() {
        this.game.draw_text(this.ctx, "*", 30, "#ffffff", this.cursorX, this.cursorY)
    }
}

class MainMenu extends Menu {
    constructor(game, ctx) {
        super(game, ctx)
        this.state = "Start"

        this.startx = this.mid_w
        this.starty = this.mid_h + 0

        this.optionsx = this.mid_w
        this.optionsy = this.mid_h + 50

        this.creditsx = this.mid_w
        this.creditsy = this.mid_h + 100

        this.quitx = this.mid_w
        this.quity = this.mid_h + 150
        this.cursorX = (this.startx + this.offset)
        this.cursorY = this.starty
    }

    display_menu() {
        //Mostra il menu e richiama il metodo che controlla se ci sono eventi
        this.run_display = true
        while (this.run_display) {
            //TODO Per non occupare moolta CPU faccio meno giri al secondo, abbasso gli fps

            //this.game.check_events()
            //this.check_input()
            this.ctx.clearRect(0, 0, this.game.canvas.width, this.game.canvas.height)
            this.ctx.fillStyle = "#0080FF" //(this.AZZURRO)
            this.ctx.fillRect(0, 0, this.game.canvas.width, this.game.canvas.height)

            this.game.draw_text(this.ctx, 'CRAZY GOOSE', 70, "#ffffff", this.game.DISPLAY_W / 2, this.game.DISPLAY_H / 2 - 200)
            this.game.draw_text(this.ctx, "Start Game", 30, "#ffffff", this.startx, this.starty)
            this.game.draw_text(this.ctx, "Options", 30, "#ffffff", this.optionsx, this.optionsy)
            this.game.draw_text(this.ctx, "Credits", 30, "#ffffff", this.creditsx, this.creditsy)
            this.game.draw_text(this.ctx, "Quit", 30, "#ffffff", this.quitx, this.quity)
            this.draw_cursor()
        }
    }
}

class OptionsMenu extends Menu {
    constructor(game, ctx) {
        super(game, ctx)
        this.state = 'Volume'
        this.volx = this.mid_w
        this.voly = this.mid_h + 20
        this.controlsx = this.mid_w
        this.controlsy = this.mid_h + 40
        this.cursorX = (this.volx + this.offset)
        this.cursorY = this.voly - 50
    }

    //cosa esce quando entri in Options Menu
    display_menu() {
        this.run_display = true
        while (this.run_display) {
            //this.game.check_events()
            // this.check_input()
            if (this.game.START_KEY || this.game.BACK_KEY) {
                this.game.curr_menu = this.game.main_menu
            }
            this.ctx.clearRect(0, 0, this.game.canvas.width, this.game.canvas.height)
            this.ctx.fillStyle = "#0080FF" //(this.AZZURRO)
            this.ctx.fillRect(0, 0, this.game.canvas.width, this.game.canvas.height)

            this.game.draw_text(this.ctx, 'Options', 70, "#ffffff", this.game.DISPLAY_W / 2, this.game.DISPLAY_H / 2 - 150)
            this.game.draw_text(this.ctx, 'Future Release', 30, "#ffffff", this.volx, this.voly)
            this.draw_cursor()
        }
    }
}

class CreditsMenu extends Menu {
    constructor(game, ctx) {
        super(game, ctx)
    }

    //schermata dei Credits
    display_menu() {
        this.run_display = true
        while (this.run_display) {
            //this.game.check_events()
            if (this.game.START_KEY || this.game.BACK_KEY) {
                this.game.curr_menu = this.game.main_menu
                this.run_display = false
            }
            this.ctx.clearRect(0, 0, this.game.canvas.width, this.game.canvas.height)
            this.ctx.fillStyle = "#0080FF" //(this.AZZURRO)
            this.ctx.fillRect(0, 0, this.game.canvas.width, this.game.canvas.height)

            this.game.draw_text(this.ctx, 'Credits', 70, "#ffffff", this.game.DISPLAY_W / 2, this.game.DISPLAY_H / 2 - 180)
            this.game.draw_text(this.ctx, "Made by Crazy Goose Team:", 30, "#ffffff", this.game.DISPLAY_W / 2, this.game.DISPLAY_H / 2 - 120)
            this.game.draw_text(this.ctx, "Bellone Giulio", 30, "#ffffff", this.game.DISPLAY_W / 2, this.game.DISPLAY_H / 2 - 30)
            this.game.draw_text(this.ctx, "Calzia Mattia", 30, "#ffffff", this.game.DISPLAY_W / 2, this.game.DISPLAY_H / 2 + 20)
            this.game.draw_text(this.ctx, "Dastrù Alessandro", 30, "#ffffff", this.game.DISPLAY_W / 2, this.game.DISPLAY_H / 2 + 70)
            this.game.draw_text(this.ctx, "Durante Andrea", 30, "#ffffff", this.game.DISPLAY_W / 2, this.game.DISPLAY_H / 2 + 120)
            this.game.draw_text(this.ctx, "Hila Kledi", 30, "#ffffff", this.game.DISPLAY_W / 2, this.game.DISPLAY_H / 2 + 170)
            this.game.draw_text(this.ctx, "Hu Qiyan", 30, "#ffffff", this.game.DISPLAY_W / 2, this.game.DISPLAY_H / 2 + 220)
            this.draw_cursor()
        }
    }
}